Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM node:8.15.0
+
+# Install VS Code's deps. These are the only two it seems we need.
+RUN apt-get update
+RUN apt-get install -y libxkbfile-dev libsecret-1-dev
+
+# Ensure latest yarn.
+RUN npm install -g yarn
+
+# In the future, we can use https://github.com/yarnpkg/rfcs/pull/53 to make it use the node_modules
+# directly which should be faster.
+WORKDIR /src
+COPY . .
+RUN yarn
+RUN yarn task build:server:binary
+
+# We deploy with ubuntu so that devs have a familiar environemnt.
+FROM ubuntu
+RUN apt-get update
+RUN apt-get install -y openssl
+WORKDIR /root/project
```

Comment:
I do not recommend using `root`. Consider using a actual user or a pseudo-user like what we do in Theia.

Reply:
I personally don't really see much value in that. Especially for a dev setup, you pretty much need sudo 100% of the time at which point it just becomes annoying and repetitive.

cc @kylecarbs

Reply:
I'm not certain what purpose there'd be in limiting the user when running inside the docker container. What do you think @sr229 ?

Reply:
@kylecarbs @nhooyr we're excercising a good practice that the user should be aware that some administrative actions might need `sudo` or a manual switch to the root. Not all Cloud IDE users are proficient in Linux and know the security implications. 

This is also how Theia and Cloud9 does their images.

Reply:
Lets get this PR in for now and discuss this in #65.